sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/NotificationHook/NotificationHook.cs

```diff
@@ -85,15 +86,88 @@ internal static HookInfoPatch GetPatchModel(NotificationHook hook)
+        {
+            if (element.TryGetProperty("hookType", out JsonElement discriminator))
+            {
+                switch (discriminator.GetString())
+                {
+                    case "Email":
+                        return EmailNotificationHook.DeserializeEmailNotificationHook(element);
+                    case "Webhook":
+                        return WebNotificationHook.DeserializeWebNotificationHook(element);
+                }
+            }
+            HookType hookType = default;
+            Optional<string> hookId = default;
+            string hookName = default;
+            Optional<string> description = default;
+            Optional<string> externalLink = default;
+            Optional<IReadOnlyList<string>> admins = default;
+            foreach (var property in element.EnumerateObject())
+            {
+                if (property.NameEquals("hookType"))
+                {
+                    hookType = new HookType(property.Value.GetString());
+                    continue;
+                }
+                if (property.NameEquals("hookId"))
+                {
+                    hookId = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("hookName"))
+                {
+                    hookName = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("description"))
+                {
+                    description = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("externalLink"))
+                {
+                    externalLink = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("admins"))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        property.ThrowNonNullablePropertyIsNull();
+                        continue;
+                    }
+                    List<string> array = new List<string>();
+                    foreach (var item in property.Value.EnumerateArray())
+                    {
+                        array.Add(item.GetString());
+                    }
+                    admins = array;
+                    continue;
+                }
+            }
+            return new UnknownNotificationHook(hookType, hookId.Value, hookName, description.Value, externalLink.Value, Optional.ToList(admins));
+        }
+
+        private class UnknownNotificationHook : NotificationHook
+        {
+            public UnknownNotificationHook(HookType hookType, string id, string name, string description, string internalExternalLink, IReadOnlyList<string> administrators)
+                : base(hookType, id, name, description, internalExternalLink, administrators)
+            {
+            }
+        }
```

Comment:
We're moving the whole serialization method to the non-generated portion of the file because we can't instantiate a `new NotificationHook(...)` anymore. If left in the generated code, it would generate compile errors.

Another question could be "why are returning an Unknown Notification Hook instead of just throwing in case the type isn't recognized?". This prevents the client from breaking if the service adds a new kind of hook in the future.

Suppose the service adds a `EventHubsHook` that's not supported by version 1.0.0 of the SDK. If a customer wants to, let's say, update the name of this hook, they would get an error:
```cs
NotificationHook hook = adminClient.GetHook(...); // we would throw here because kind is unknown.

hook.Name = "<new-name>";
adminClient.UpdateHook(hook);
```

However, updating these common properties, such as `Name`, `Description`, etc, is something we could do without really knowing the kind of hook, so that's what we're trying to support here. A new kind of hook could also break listing APIs, such as the service method `GetHooks`, which would break even if the customer is not trying to update the new hook in particular.